### PR TITLE
ITL-2

### PR DESCRIPTION
**Pull request by Balazs14-hun**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,9 +1,45 @@
...
 import HeaderComponent from '@/components/HeaderComponent.vue'
+import BaseNoTodo from '@/components/BaseNoTodo.vue'
+import type Todo from '@/types/Todo'
+import TodoList from '@/components/TodoList.vue'
+
+const todos = ref<Todo[]>([])
+
+function createTodo() {
+  todos.value.unshift({
+    id: Math.random(),
```
Comment (szabolcs3) [inline src/App.vue]: Use length of the array here please, instead of `Math.random()`

File: src/App.vue [hunk 2]
```diff
@@ -1,9 +1,45 @@
...
+    title: 'Best todo',
+    description:
+      'Do nothing, do nothing, do nothing, do nothing, do nothing, do nothing, do nothing...',
+    priority: 'Low',
+    done: false
```
Comment (szabolcs3) [inline src/App.vue]: `isChecked` is a better variable name I think 😛 . Also please use `is` prefix for boolean variables.

File: src/components/BasePriority.vue [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<template>
+  <p
+    :class="`${bgColor} w-fit place-self-end rounded-2xl px-4 text-sm text-white sm:px-5 sm:text-base`"
```
Comment (szabolcs3) [inline src/components/BasePriority.vue]: You can use here `class` and `:class` attributes.
`class` attribute will contain the constant classnames and the `:class` attribute will contain the dynamic variable.

File: src/components/TodoCard.vue [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<template>
+  <div class="grid w-72 grid-cols-2 gap-6 rounded-lg border-2 border-black p-3 sm:w-[30rem]">
+    <h1 class="text-lg font-bold sm:text-xl">{{ todo.title }}</h1>
+    <BasePriority
```
Comment (szabolcs3) [inline src/components/TodoCard.vue:4]: Please use a background mapper here, so you don't need to create 3 variant for the priority for example:
```
const backgroundMapper = {
'High' : 'bg-orange-600',
...
```

File: src/components/TodoList.vue [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <TodoCard
```
Comment (szabolcs3) [inline src/components/TodoList.vue:2]: Use `v-model` please 😄

File: src/components/TodoCard.vue [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        class="invisible absolute -top-2 size-6 stroke-green-500 stroke-2 opacity-90 peer-checked:visible sm:size-8"
+      />
+    </label>
+  </div>
+</template>
+
+<script setup lang="ts">
+import BasePriority from '@/components/BasePriority.vue'
+import { CheckIcon } from '@heroicons/vue/20/solid'
+import type Todo from '@/types/Todo'
```
Comment (szabolcs3) [inline src/components/TodoCard.vue]: Please remove the `type` keyword from here.

File: src/components/TodoList.vue [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    v-for="todo in todos"
+    :key="todo.id"
+    class="mt-7"
+    :todo="todo"
+  />
+</template>
+
+<script setup lang="ts">
+import TodoCard from '@/components/TodoCard.vue'
+import type Todo from '@/types/Todo'
```
Comment (szabolcs3) [inline src/components/TodoList.vue]: Please remove the `type` keyword from here.

File: src/types/Todo.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export default interface Todo {
+  id: number
+  title: string
+  description: string
+  priority: 'Low' | 'Medium' | 'High'
+  done: boolean
```
Comment (szabolcs3) [inline src/types/Todo.ts]: `isChecked` is a better variable name I think 😛 . Also please use `is` prefix for boolean variables.

File: src/components/TodoCard.vue [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      :bg-color="'bg-yellow-500'"
+    />
+    <BasePriority
+      v-else
+      :text="todo.priority"
+      :bg-color="'bg-teal-500'"
+    />
+    <p class="text-xs font-semibold text-slate-400 sm:text-base">{{ todo.description }}</p>
+    <label class="relative cursor-pointer place-self-end">
+      <input
```
Comment (szabolcs3) [inline src/components/TodoCard.vue]: Please create a separate component for the checkbox.

**Review by szabolcs3 — CHANGES_REQUESTED** [review]

File: src/App.vue [hunk 10]
```diff
@@ -1,9 +1,45 @@
...
+    />
+    <BaseNoTodo v-else />
   </div>
 </template>
 
 <script setup lang="ts">
+import { ref } from 'vue'
 import HeaderComponent from '@/components/HeaderComponent.vue'
+import BaseNoTodo from '@/components/BaseNoTodo.vue'
+import type Todo from '@/types/Todo'
```
Comment (zsolt-wbg) [inline src/App.vue]: Remove type from here pls

File: src/components/BaseNoTodo.vue [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+  <div class="flex flex-col pt-10">
+    <img
+      class="size-64 sm:size-80"
+      src="@/assets/svg/no-todo.svg"
+    />
+    <p class="pt-5 text-xl font-semibold text-slate-500 sm:text-4xl">You have no todos yet</p>
+  </div>
+</template>
+
+<script setup lang="ts"></script>
```
Comment (zsolt-wbg) [inline src/components/BaseNoTodo.vue]: You can remove script part if you don't use it

File: src/components/TodoCard.vue [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+      :text="todo.priority"
+      :bg-color="'bg-yellow-500'"
+    />
+    <BasePriority
+      v-else
+      :text="todo.priority"
+      :bg-color="'bg-teal-500'"
+    />
+    <p class="text-xs font-semibold text-slate-400 sm:text-base">{{ todo.description }}</p>
+    <label class="relative cursor-pointer place-self-end">
```
Comment (zsolt-wbg) [inline src/components/TodoCard.vue]: Please use div instead of label

File: src/components/TodoList.vue [hunk 13]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <TodoCard
+    v-for="todo in todos"
+    :key="todo.id"
+    class="mt-7"
```
Comment (zsolt-wbg) [inline src/components/TodoList.vue]: Pay attention to the attribute order :)

File: src/types/Todo.ts [hunk 14]
```diff
@@ -0,0 +1,7 @@
+export default interface Todo {
+  id: number
+  title: string
+  description: string
+  priority: 'Low' | 'Medium' | 'High'
```
Comment (zsolt-wbg) [inline src/types/Todo.ts]: Add Priority as a separate type

File: src/types/Todo.ts [hunk 15]
```diff
@@ -0,0 +1,7 @@
+export default interface Todo {
```
Comment (zsolt-wbg) [inline src/types/Todo.ts]: You don't need _default_ here

**Review by zsolt-wbg — CHANGES_REQUESTED** [review]

**Review by zsolt-wbg — APPROVED** [review]

File: src/App.vue [hunk 16]
```diff
@@ -1,9 +1,45 @@
...
+import { ref } from 'vue'
 import HeaderComponent from '@/components/HeaderComponent.vue'
+import BaseNoTodo from '@/components/BaseNoTodo.vue'
+import { Todo } from '@/types/Todo'
+import TodoList from '@/components/TodoList.vue'
+
+const todos = ref<Todo[]>([])
+
+function createTodo() {
+  todos.value.unshift({
```
Comment (szabolcs3) [inline src/App.vue:22]: Please create a new todo (priority should be low and isChecked false by default)

**Review by szabolcs3 — CHANGES_REQUESTED** [review]

**Pull request merged**